node/src/server_initializer.rs

```diff
@@ -112,20 +80,24 @@ impl Future for ServerInitializer {
     }
 }
 
-impl ServerInitializer {
-    pub fn new() -> ServerInitializer {
-        ServerInitializer {
+impl Default for ServerInitializerReal {
+    fn default() -> ServerInitializerReal {
+        ServerInitializerReal {
             dns_socket_server: Box::new(DnsSocketServer::new()),
             bootstrapper: Box::new(Bootstrapper::new(Box::new(LoggerInitializerWrapperReal {}))),
             privilege_dropper: Box::new(PrivilegeDropperReal::new()),
+            dirs_wrapper: Box::new(DirsWrapperReal),
         }
     }
+}
+
+trait CombineResults {
```

Comment:
The name of a trait shouldn't be a verb. Mostly it should be a noun. For example, ResultsCombiner. (Maybe there's a better one, but that occurs to me immediately, which means it might immediately mean something to somebody else.)